Cython/Debugger/Cygdb.py

```diff
@@ -104,67 +104,62 @@ def make_command_file(path_to_debug_info, prefix_code='',
 
     return tempfilename
 
-usage = "Usage: cygdb [options] [PATH [-- GDB_ARGUMENTS]]"
 
-def main(path_to_debug_info=None, gdb_argv=None, no_import=False):
+def main():
     """
     Start the Cython debugger. This tells gdb to import the Cython and Python
     extensions (libcython.py and libpython.py) and it enables gdb's pending
     breakpoints.
-
-    path_to_debug_info is the path to the Cython build directory
-    gdb_argv is the list of options to gdb
-    no_import tells cygdb whether it should import debug information
     """
-    parser = optparse.OptionParser(usage=usage)
-    parser.add_option("--gdb-executable",
+    parser = argparse.ArgumentParser(
+        prog="cygdb",
+        description="Cython debugger",
+    )
+    parser.add_argument("gdb_argv", nargs="*",
+                        help="arguments to forward to gdb; specified after --")
+    parser.add_argument("--build_dir", default=os.curdir,
+                        help="directory containing cython_build/ files")
+    parser.add_argument("--no_import", default=False,
+                        help="whether cygdb should import debug information")
+    parser.add_argument("--gdb-executable",
         dest="gdb", default='gdb',
         help="gdb executable to use [default: gdb]")
-    parser.add_option("--verbose", "-v",
+    parser.add_argument("--verbose", "-v",
         dest="verbosity", action="count", default=0,
         help="Verbose mode. Multiple -v options increase the verbosity")
-    parser.add_option("--skip-interpreter",
+    parser.add_argument("--skip-interpreter",
                       dest="skip_interpreter", default=False, action="store_true",
                       help="Do not automatically point GDB to the same interpreter "
                            "used to generate debugging information")
 
-    (options, args) = parser.parse_args()
-    if path_to_debug_info is None:
-        if len(args) > 1:
-            path_to_debug_info = args[0]
-        else:
-            path_to_debug_info = os.curdir
-
-    if gdb_argv is None:
-        gdb_argv = args[1:]
+    parsed = parser.parse_args()
+    path_to_debug_info = parsed.build_dir
+    gdb_argv = parsed.gdb_argv
+    no_import = parsed.no_import
 
     if path_to_debug_info == '--':
```

Comment:
Is it possible that parameter `build_dir` will contain `--`. I suppose `no_import` variable is controlled by `--no-import` argument, so this can be removed.

Reply:
Yea I agree it makes more sense to just control that from the command line arguments